### ISSUE TITLE
bug: rule with a space in the name breaks Rules dashboard

### ISSUE THREAD

**Issue by johnarnold** [StackStorm/st2web#375]
Repro:

In the Rule editor, create a rule with a space in the name, and save it.

Result:
```
TypeError: Cannot read property 'getRef' of null
    at https://100.127.8.138/js/apps.js:2:22790
    at Function.Hn (https://100.127.8.138/node_modules/lodash/dist/lodash.js:37:156)
    at g.o.(anonymous function) [as filter] (https://100.127.8.138/node_modules/lodash/dist/lodash.js:62:438)
    at https://100.127.8.138/js/apps.js:2:22748
    at <anonymous>
$delegate.criticalError @ modules.js:4
```
The Rules dashboard loads, but if you click on the rule, it will dump you to the History dashboard.

**Comment by LindsayHill** [comment]
@johnarnold which version of ST2 are you using? I believe this should be fixed in 2.3.1 https://github.com/StackStorm/st2web/issues/357

**Comment by johnarnold** [comment]
```
# st2 --version
st2 2.3.1
```

**Comment by johnarnold** [comment]
ref https://github.com/StackStorm/st2/issues/2004

**Comment by christopherjbaker** [comment]
This has been fixed in the [react rewrite](https://github.com/StackStorm/st2web/pull/420), which should be included with 2.6.
